src/f5_tts/model/duration.py

```diff
@@ -0,0 +1,358 @@
+# adapted from https://github.com/lucasnewman/f5-tts-mlx/blob/main/f5_tts_mlx/duration.py
```

Comment:
duration class should go into modules.py
and leverage existing modules there (cuz it's all the same for attn, ffn, melspec, pos_emb, etc.)